lib/ansible/runner.rb

```diff
@@ -237,6 +252,23 @@ def validate_params!(env_vars, extra_vars, tags, ansible_runner_method, playbook
         raise ArgumentError, errors.join("; ") if errors.any?
       end
 
+      def credentials_info(credentials, base_dir)
+        command_line = {}
+        env_vars     = {}
+        extra_vars   = {}
+        credentials.each do |id|
+          cred = Ansible::Runner::Credential.new(id, base_dir)
```

Comment:
The performance side of me cringes at this because this is technically an N+1 here (since we are fetching each credential type one by one in the base class), but I know your response will be "N will never be that large, so it doesn't matter".

Mostly saying this so I can be #technicallyCorrectOnTheInternet™ and boost my internet points score.

Reply:
Yeah ... I noticed that, but I would rather this than put all that logic into the model.